### ISSUE TITLE
Failure to use less from BusyBox

### ISSUE THREAD

**Issue by hungpham3112** [sharkdp/bat#2143]
**Describe the bug you encountered:**
Step to reproduce:
1. scoop install busybox, bat
2. bat .\yourfile.

Here is the result in my machine:

![image](https://user-images.githubusercontent.com/75968004/161364026-bae24516-5efa-4650-985c-4b284c02cace.png)

**What did you expect to happen instead?**

No error

**How did you install `bat`?**

scoop install bat

---

**bat version and environment**

version: bat 0.20.0 (0655ecf)
OS: Window 11 lastest

**Comment by Enselic** [comment]
Thank you for the taking the time to report a bug. But there is not much the bat project can do about this. Please try to reach out to the scoop packagers of bat.

**Comment by hungpham3112** [comment]
> Thank you for the taking the time to report a bug. But there is not much the bat project can do about this. Please try to reach out to the scoop packagers of bat.

I don't know why but without busybox, it runs normal.

**Comment by Enselic** [comment]
Probably busybox packages a bin called bat for some reason. In any case, let's close this particular issue in this tracker.

**Comment by keith-hall** [comment]
Try with `--paging=never` or setting `BAT_PAGER` to override the command line args passed to `less` - it looks likely that busybox includes some dodgy version of `less`

**Comment by Enselic** [comment]
Great save @keith-hall, that is most likely what is going on here 👍

**Comment by Enselic** [comment]
Actually, we already do quite extensive version checks on less when bat starts. Perhaps we could try to catch this case too and print a warning or something. I suggest we re-open this. Might also be suitable for a new contributor could take a look at.

**Comment by nfisher1226** [comment]
I'd like to take this one on since nobody else has claimed it.

**Comment by keith-hall** [comment]
@nfisher1226 sounds great, thanks. Please let us know if you need any pointers :)

**Comment by nfisher1226** [comment]
So looking at busybox less as it gets built with the default config, there's no color support, but it can handle bat's other formatting decorations (in somewhat reduced capacity). Example:
```
───────┬──────────────────────────────────────────────────────────────────────��
�─
       │ File: applets/applets.c
       │ Size: 350 B
───────┼──────────────────────────────────────────────────────────────────────��
�─
   1   │ /* vi: set sw=4 ts=4: */
   2   │ /*
   3   │  * Stub for linking busybox binary against libbusybox.
   4   │  *
   5   │  * Copyright (C) 2007 Denys Vlasenko <vda.linux@googlemail.com>
   6   │  *
   7   │  * Licensed under GPLv2, see file LICENSE in this source tree.
   8   │  */
   9   │ #include "busybox.h"
  10   │ 
  ...
So in less.rs the `parse_less_version` returns an `Option<usize>` due to less's unconventional versioning, but busybox is on semantic versioning, so that return type gets messy if we try to account for busybox. It seems like we're only doing that from output.rs (based on a quick `grep -R` in the source tree), so here's my initial thinking.
* Return an enum instead, something like:
```Rust
pub enum LessVersion {
    less(usize),
    busybox,
    unknown,
```
* Only turn off colored output for busybox less, but leave the other decorations. They're kind of ugly, but usable, and there might well be environments where busybox less is the only pager - say Alpine probably.
If that sounds reasonable I can get something put together pretty quickly. I see a couple of tests that might affect, too, so I can take a look at those at the same time.

### Questions
* Should I check older versions of busybox for compatability?
* If so, how far back?
* The horizontal grid lines are ugly with busybox less. How much do we care? Enough to change how they're drawn or possibly disable them? Or just let it go.

**Comment by nfisher1226** [comment]
Ok, thanks for the feedback.

**Comment by nfisher1226** [comment]
Initial draft in https://github.com/sharkdp/bat/pull/2162

**Comment by nfisher1226** [comment]
So in my draft it looks like the CI failed due to my changing `tests/syntax-tests/source/Rust/output.rs`. I'm guessing that file is only being used for testing syntax highlighting and I should have left it alone?

**Comment by keith-hall** [comment]
That would work I think, yes :) I also commented on the PR with another solution to fix CI. I guess it could make sense to keep the syntax test file in sync, but not too important

**Comment by nfisher1226** [comment]
I see your comment. I'll try running that update script. Thanks.

**Comment by Enselic** [comment]
Closed by #2162

**Comment by eggbean** [comment]
I just came across the same issue with `BusyBox v1.36` and `bat 0.21.0` (installed through scoop).

But there is not any problem with the `less` package from scoop, so a my solution is:

    > scoop install less
    > scoop shim alter less

And select the `less` option.
